tests/python_package_test/test_dask.py

```diff
@@ -805,6 +808,67 @@ def f(part):
         assert 'foo' in str(info.value)
 
 
+@pytest.mark.parametrize('task,output', tasks_outputs)
+def test_training_having_workers_without_data(client, task, output):
```

Comment:
```suggestion
def test_training_succeeds_even_if_some_workers_do_not_have_any_daata(client, task, output):
```

I'd prefer a more informative name here.